### ISSUE TITLE
Compress functionality hangs with nested object arguments

### ISSUE THREAD

**Issue by imolorhe** [rse/graphql-query-compress#8]
As reported in https://github.com/imolorhe/altair/issues/247, there is a bug with the compress functionality when used with nested object arguments like:

```
mutation{
  updateProducts(payloads: [{
    creative_id: "3333",
    product_type_slug: "print",
    scores: {
      sales_hotness_product: {
        
      }
    }
  }]){
    error
    error_count
  }
}
```

**Comment by rse** [comment]
Sorry, I do not see any bug if I enter your sample into my `sample.js` file:

```
$ cat sample.js 
const compress = require(".")
let query = `mutation {
    updateProducts(payloads: [ {
        creative_id: "3333",
        product_type_slug: "print",
        scores: {
            sales_hotness_product: {
            }
        }
    } ] ) {
        error
        error_count
    }
}`
query = compress(query)
console.log(query)
$ node sample.js 
mutation{updateProducts(payloads:[{creative_id:"3333",product_type_slug:"print",scores:{sales_hotness_product:{}}}]){error error_count}}
```
Your query is compressed just fine and without problems IMHO.

**Comment by imolorhe** [comment]
Could it be an issue with the version?

**Comment by imolorhe** [comment]
I've just tried again using `latest` version but it still freezes when I try that query.

**Comment by rse** [comment]
Does your application freezes or does really graphql-query-compress freeze? As you can see in my test above, if I enter your particular GraphQL query into the sample.js from graphql-query-compress, nothing freezes and the correct output results, too. If something freezes, I guess it is your further processing inside your application and not graphql-query-compress.

**Comment by imolorhe** [comment]
@rse I created a simple test repo to verify this. It turns out there's some weird behavior going on.
Two queries that are practically the same, one of them is compressed successfully, the second one hangs. 😕 

`query` compresses successfully, `query2` hangs.
https://github.com/imolorhe/graphql-compress-test/blob/master/index.js

**Comment by rse** [comment]
Yes, you're right: the query2, although it seems just have a different formatting, hangs on compression. Very interesting. I'll investigate...

**Comment by imolorhe** [comment]
Thanks.

**Comment by rse** [comment]
OK, I've found the origin of the trouble. There was a regexp for matching the double-quoted strings which had two qualifiers in sequence and this causes the exponential run-time (seen as a visible hang, but it actually tried to match the regexp). Now fixed in version 0.9.10. Thanks for your feedback.
